tests/credentials_test.go

```diff
@@ -320,70 +224,90 @@ var _ = Describe("[sig-compute]Guest Access Credentials", decorators.SigCompute,
 				&expect.BExp{R: "test-ssh-key2"},
 				&expect.BSnd{S: "cat /home/fedora/.ssh/authorized_keys\n"},
 				&expect.BExp{R: "test-ssh-key3"},
-			}, time.Second*180)
+			}, time.Second*180)).To(Succeed())
 		}
 
 		BeforeEach(func() {
-			vmi = tests.NewRandomVMIWithEphemeralDiskHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling))
-			vmi.Namespace = util.NamespaceTestDefault
-
 			By("Creating a secret with three ssh keys")
-			secret := kubev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretID,
-					Namespace: vmi.Namespace,
-					Labels: map[string]string{
-						util.SecretLabel: secretID,
-					},
-				},
-				Type: "Opaque",
-				Data: map[string][]byte{
-					"my-key1": []byte(key1),
-					"my-key2": []byte(key2),
-					"my-key3": []byte(key3),
-				},
-			}
-			_, err := virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), &secret, metav1.CreateOptions{})
-			Expect(err).ToNot(HaveOccurred())
+			createNewSecret(secretID, map[string][]byte{
+				"my-key1": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"),
+				"my-key2": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"),
+				"my-key3": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"),
+			})
+
 		})
 
-		It("[test_id:TODO]should have ssh-key under authorized keys added by NoCloud", func() {
-			tests.AddCloudInitNoCloudData(vmi, "disk1", userData, "", false)
-			vmi.Spec.AccessCredentials = []v1.AccessCredential{
-				{
-					SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
-						Source: v1.SSHPublicKeyAccessCredentialSource{
-							Secret: &v1.AccessCredentialSecretSource{
-								SecretName: secretID,
+		When("using configdrive", func() {
```

Comment:
with the global `libvmi.func WithAccessCredentials(secretName string, propagationMethod v1.SSHPublicKeyAccessCredentialPropagationMethod)` function `When("using configdrive"...` and `When("using nocloud"...` specs can be reduced as
```
DescribeTable("should have ssh-key under authorized keys added", func(propagationMethod v1.SSHPublicKeyAccessCredentialPropagationMethod) {
			vmi := libvmi.NewFedora(
				libvmi.WithCloudInitConfigDriveData(userData, false),
				libvmi.WithAccessCredentials(secretID, propagationMethod))
			vmi = tests.RunVMIAndExpectLaunch(vmi, fedoraRunningTimeout)
			verifySSHKeys(vmi)
		},
			Entry("[test_id:6224] using configdrive", v1.SSHPublicKeyAccessCredentialPropagationMethod{
				ConfigDrive: &v1.ConfigDriveSSHPublicKeyAccessCredentialPropagation{},
			}),
			Entry("[test_id:TODO] using nocloud", v1.SSHPublicKeyAccessCredentialPropagationMethod{
				NoCloud: &v1.NoCloudSSHPublicKeyAccessCredentialPropagation{},
			}),
		)
```
Just an idea. WDYT?

Reply:
Done